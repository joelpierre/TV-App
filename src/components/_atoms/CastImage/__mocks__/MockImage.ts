import { ITvShowImage } from '../../../../types/interfaces';

export const MockImage: ITvShowImage = {
  medium: "http://static.tvmaze.com/uploads/images/medium_portrait/130/325669.jpg",
  original: "http://static.tvmaze.com/uploads/images/original_untouched/130/325669.jpg"
};
