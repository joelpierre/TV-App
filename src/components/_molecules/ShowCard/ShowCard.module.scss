.ShowCard {
  $root: &;
  display: block;
  width: 100%;
  background-color: $color-tint-alpha;

  @include hover-focus {
    text-decoration: none;
    color: $color-brand;

    #{$root}__content {
      border-bottom-color: $color-brand;
    }
  }

  &__content {
    display: flex;
    flex-direction: column-reverse;
    border-bottom: 2px solid transparent;
    transition: border-bottom-color .4s;
  }

  &__heading {
    font-family: $font-medium;
    font-size: 16px;
    margin-top: spacing(1.5);
    margin-bottom: spacing(1.5);

    @include breakpoint('sm') {
      margin-top: 0;
    }
  }

  &__meta {
    font-size: 14px;
    margin-bottom: spacing(0.5);

    &:last-of-type {
      margin-bottom: spacing(0.5);
    }
  }

  &__meta-time {
    font-size: 12px;
  }

  &__ratings {
    margin-top: spacing(2);
    margin-bottom: spacing();
  }

  &__ratings,
  &__meta {
    display: none;

    @include breakpoint('sm') {
      display: block;
    }
  }

  &__copy,
  &__meta,
  &__heading {
    transition: color .4s;
  }
}
