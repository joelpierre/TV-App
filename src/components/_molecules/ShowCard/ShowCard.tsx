import classNames from 'classnames';
import React from 'react';

import styles from './ShowCard.module.scss';

interface IShowCardProps {
  className?: string;
}

const ShowCard: React.FunctionComponent<IShowCardProps> = ({
  className
}) => {
  return (
    <section className={classNames(styles.ShowCard, className)}>
      This is a show card
    </section>
  );
};

export default ShowCard;
