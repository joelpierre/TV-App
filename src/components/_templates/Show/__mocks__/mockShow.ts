/* eslint-disable */
import { ITvShow } from '../../../../types/interfaces';
import { MockImage } from '../../../_atoms/CastImage/__mocks__/MockImage';

export const MockShow: ITvShow = {
  "id": 31534,
  "url": "http://www.tvmaze.com/shows/31534/the-russell-howard-hour",
  "name": "The Russell Howard Hour",
  "type": "Talk Show",
  "language": "English",
  "genres": [
    "Comedy"
  ],
  "status": "Running",
  "runtime": 60,
  "premiered": "2017-09-21",
  "officialSite": "https://www.sky.com/watch/title/series/89419c88-0079-426a-b29b-bc28bc403a58/the-russell-howard-hour",
  "schedule": {
    "time": "22:00",
    "days": [
      "Thursday"
    ]
  },
  "rating": {
    "average": 8.4
  },
  "weight": 83,
  "network": {
    "id": 63,
    "name": "Sky 1",
    "country": {
      "name": "United Kingdom",
      "code": "GB",
      "timezone": "Europe/London"
    }
  },
  "webChannel": null,
  "externals": {
    "tvrage": null,
    "thetvdb": 334902,
    "imdb": "tt7544016"
  },
  "image": MockImage,
  "summary": "<p><b>The Russell Howard Hour</b> is a weekly talk show in which Russell Howard will be chatting with celebrity guests alongside comedy stand-up. </p>",
  "updated": 1603404686,
  "_links": {
    "self": {
      "href": "http://api.tvmaze.com/shows/31534"
    },
    "previousepisode": {
      "href": "http://api.tvmaze.com/episodes/1955660"
    },
    "nextepisode": {
      "href": "http://api.tvmaze.com/episodes/1955661"
    }
  },
  "_embedded": {
    "cast": [
      {
        "person": {
          "id": 64866,
          "url": "http://www.tvmaze.com/people/64866/russell-howard",
          "name": "Russell Howard",
          "country": {
            "name": "United Kingdom",
            "code": "GB",
            "timezone": "Europe/London"
          },
          "birthday": "1980-03-23",
          "deathday": null,
          "gender": "Male",
          "image": {
            "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/175/437607.jpg",
            "original": "http://static.tvmaze.com/uploads/images/original_untouched/175/437607.jpg"
          },
          "_links": {
            "self": {
              "href": "http://api.tvmaze.com/people/64866"
            }
          }
        },
        "character": {
          "id": 456592,
          "url": "http://www.tvmaze.com/characters/456592/the-russell-howard-hour-host",
          "name": "Host",
          "image": null,
          "_links": {
            "self": {
              "href": "http://api.tvmaze.com/characters/456592"
            }
          }
        },
        "self": true,
        "voice": false
      }
    ]
  }
};
