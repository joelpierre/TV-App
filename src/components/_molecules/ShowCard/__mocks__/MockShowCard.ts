import { ITvSchedule } from '../../../../types/interfaces';
import { MockShow } from '../../../_templates/Show/__mocks__/mockShow';
import { MockImage } from '../../../_atoms/CastImage/__mocks__/MockImage';

export const MockShowCard: ITvSchedule = {
  url: '',
  type: '',
  summary: '',
  season: 2,
  runtime: 50,
  number: 2,
  name: 'Test Show Card',
  image: MockImage.original,
  id: 123,
  airtime: '09:00',
  airstamp: 'airStamp',
  airdate: '01-01-2020',
  show: MockShow
};
