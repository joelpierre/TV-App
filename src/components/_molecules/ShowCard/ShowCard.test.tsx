import React from 'react';
import { render, RenderResult } from 'test-utils';

import ShowCard, { IShowCardProps } from './ShowCard';
import { MockShowCard } from '@molecules/ShowCard/__mocks__/MockShowCard';

const defaultProps: IShowCardProps = {
  ...MockShowCard
};

const setup = (props: Partial<IShowCardProps> = {}): RenderResult => {
  const setupProps = { ...defaultProps, ...props };
  return render(<ShowCard {...setupProps} />);
};

describe('<ShowCard/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
