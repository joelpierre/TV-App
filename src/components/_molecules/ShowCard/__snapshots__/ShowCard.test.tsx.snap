// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowCard/> should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="Anchor ShowCard"
    href="/show/31534"
  >
    <article
      class="ShowCard__content"
    >
      <p
        class="ShowCard__meta"
      >
        Time: 09:00 
        <span
          class="ShowCard__meta-time"
        >
          (24hr)
        </span>
      </p>
      <p
        class="ShowCard__meta"
      >
        Network: Sky 1
      </p>
      <h3
        class="ShowCard__heading"
      >
        The Russell Howard Hour
      </h3>
      <div
        class="Ratings ShowCard__ratings"
      >
        <span
          class="Star Ratings__star Star--is-active"
        />
        <span
          class="Star Ratings__star Star--is-active"
        />
        <span
          class="Star Ratings__star Star--is-active"
        />
        <span
          class="Star Ratings__star Star--is-active"
        />
        <span
          class="Star Ratings__star"
        />
      </div>
      <div
        class="ShowImage ShowCard__image"
      >
        <img
          alt="The Russell Howard Hour poster"
          class="ShowImage__image"
          loading="lazy"
          src="//static.tvmaze.com/uploads/images/medium_portrait/130/325669.jpg"
          width="200"
        />
      </div>
    </article>
  </a>
</DocumentFragment>
`;
